src/routes/v1/logs_stream.go

```diff
@@ -0,0 +1,115 @@
+package v1
+
+import (
+	"fmt"
+	"github.com/dana-team/platform-backend/src/controllers"
+	websocketpkg "github.com/dana-team/platform-backend/src/websocket"
+	"github.com/gin-gonic/gin"
+	"github.com/gorilla/websocket"
+	"go.uber.org/zap"
+	"io"
+	"k8s.io/client-go/kubernetes"
+	"net/http"
+)
+
+const (
+	namespaceParam      = "namespace"
+	cappNameParam       = "name"
+	containerQueryParam = "container"
+	podNameQueryParam   = "podName"
+)
+
+// GetPodLogs returns a handler function that fetches logs for a specified pod and container.
+func GetPodLogs() gin.HandlerFunc {
+	return createLogHandler(streamPodLogs, podNameQueryParam, "Pod")
+}
+
+// GetCappLogs returns a handler function that fetches logs for a specified Capp.
+func GetCappLogs() gin.HandlerFunc {
+	return createLogHandler(streamCappLogs, cappNameParam, "Capp")
+}
+
+// createLogHandler creates a gin.HandlerFunc for streaming logs using the provided stream function.
+func createLogHandler(streamFunc func(*gin.Context, *zap.Logger) (io.ReadCloser, error), paramKey, logPrefix string) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		if !websocket.IsWebSocketUpgrade(c.Request) {
+			c.AbortWithStatus(http.StatusBadRequest)
+			return
+		}
+
+		websocketClient := websocketpkg.NewWebSocket(nil)
+		conn, err := websocketClient.Register(c)
+		if err != nil {
+			websocketpkg.SendErrorMessage(conn, "error registering WebSocket")
```

Comment:
Because in WebSocket connections we send error messages over the open connection and ensuring its proper closure, whereas for HTTP routes, we use a request-response model with status codes and messages, and the connection closes automatically after the response.